core/src/main/java/io/questdb/cairo/mv/MatViewRefreshJob.java

```diff
@@ -280,7 +384,8 @@ private SampleByIntervalIterator findSampleByIntervals(
                     .$(", iteratorMaxTs<").$ts(iteratorMaxTs)
                     .I$();
 
-            return intervalIterator;
+            refreshIntervals.intervalIterator = intervalIterator;
+            return refreshIntervals;
         }
 
         return null;
```

Comment:
We need to bump the base table txn with every run, if it changes, to let the WAL Purge job clean the WAL segments of the base table.

Reply:
Done in 86cf49aa76353403da8087df3e593ca42d022b7b